Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -289,46 +296,46 @@ std::string DecodeString(pb_istream_t* stream) {
 // TODO(rsgowman): Refactor to use a helper class that wraps the stream struct.
 // This will help with error handling, and should eliminate the issue of two
 // 'EncodeFieldValue' methods.
-Status EncodeFieldValueImpl(Writer* writer, const FieldValue& field_value) {
+void EncodeFieldValueImpl(Writer* writer, const FieldValue& field_value) {
   // TODO(rsgowman): some refactoring is in order... but will wait until after a
   // non-varint, non-fixed-size (i.e. string) type is present before doing so.
+  if (!writer->status().ok()) return;
```

Comment:
If `Write` methods are no-ops, I don't think this early `return` is necessary?

Reply:
Done.